pilot/pkg/model/push_context.go

```diff
@@ -261,44 +256,12 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 
 	// Do not merge when any one is empty
 	if len(first.ConfigsUpdated) > 0 && len(other.ConfigsUpdated) > 0 {
-		merged.ConfigsUpdated = make(map[resource.GroupVersionKind]map[string]struct{})
-		for kind := range first.ConfigsUpdated {
-			merged.ConfigsUpdated[kind] = make(map[string]struct{})
-		}
-		for kind := range other.ConfigsUpdated {
-			if _, exists := merged.ConfigsUpdated[kind]; !exists {
-				merged.ConfigsUpdated[kind] = make(map[string]struct{})
-			}
+		merged.ConfigsUpdated = make(map[ConfigKey]struct{}, len(first.ConfigsUpdated)+len(other.ConfigsUpdated))
+		for conf := range first.ConfigsUpdated {
+			merged.ConfigsUpdated[conf] = struct{}{}
 		}
-
-		if !merged.Full {
-			for kind := range merged.ConfigsUpdated {
-				d1 := first.ConfigsUpdated[kind]
-				d2 := other.ConfigsUpdated[kind]
-
-				for update := range d1 {
-					merged.ConfigsUpdated[kind][update] = struct{}{}
-				}
-
-				for update := range d2 {
-					if _, exists := merged.ConfigsUpdated[kind][update]; !exists {
-						merged.ConfigsUpdated[kind][update] = struct{}{}
-					}
-				}
-			}
-		}
-	}
-
-	// Merge the target namespaces
-	if len(first.NamespacesUpdated) > 0 && len(other.NamespacesUpdated) > 0 {
-		merged.NamespacesUpdated = make(map[string]struct{})
-		for update := range first.NamespacesUpdated {
-			merged.NamespacesUpdated[update] = struct{}{}
-		}
-		for update := range other.NamespacesUpdated {
-			if _, exists := merged.NamespacesUpdated[update]; !exists {
-				merged.NamespacesUpdated[update] = struct{}{}
-			}
+		for conf := range other.ConfigsUpdated {
+			merged.ConfigsUpdated[conf] = struct{}{}
```

Comment:
nit:
```
if _, exists := merged. ConfigsUpdated[conf]; !exists {
				merged.ConfigsUpdated[update] = struct{}{}
			}
```

Reply:
An extra check-existence does seem to improve a lot but make the code more complex.